src/System.Security.Cryptography.Csp/tests/PasswordDeriveBytesTests.cs

```diff
@@ -0,0 +1,452 @@
+                Assert.NotEqual(saltIn, deriveBytes.Salt);
+            }
+        }
+
+        [Fact]
+        public static void GetSaltCopies()
+        {
+            byte[] first;
+            byte[] second;
+
+            using (var deriveBytes = new PasswordDeriveBytes(TestPassword, s_testSalt, "SHA1", DefaultIterationCount))
+            {
+                first = deriveBytes.Salt;
+                second = deriveBytes.Salt;
+            }
+
+            Assert.NotSame(first, second);
+            Assert.Equal(first, second);
+        }
+
+        [Fact]
+        public static void SetSaltAfterGetBytes_Throws()
+        {
+            using (var deriveBytes = new PasswordDeriveBytes(TestPassword, s_testSalt))
+            {
+                deriveBytes.GetBytes(1);
+                Assert.Throws<CryptographicException>(() => deriveBytes.Salt = s_testSalt);
+            }
+        }
+
+        [Fact]
+        public static void SetSaltAfterGetBytes_Reset()
+        {
+            using (var deriveBytes = new PasswordDeriveBytes(TestPassword, s_testSalt))
+            {
+                deriveBytes.GetBytes(1);
+                deriveBytes.Reset();
+                deriveBytes.Salt = s_testSaltB;
+                Assert.Equal(s_testSaltB, deriveBytes.Salt);
+            }
+        }
+
+        [Fact]
+        public static void MinimumAcceptableInputs()
+        {
+            byte[] output;
+
+            using (var deriveBytes = new PasswordDeriveBytes(string.Empty, new byte[8], "SHA1", 1))
+            {
+                output = deriveBytes.GetBytes(1);
+            }
+
+            Assert.Equal(1, output.Length);
+            Assert.Equal(0xF8, output[0]);
+        }
+
+        [Fact]
+        public static void GetBytes_ZeroLength()
+        {
+            using (var deriveBytes = new PasswordDeriveBytes(TestPassword, s_testSalt))
+            {
+                Assert.Throws<ArgumentException>(() => deriveBytes.GetBytes(0));
+            }
+        }
+
+        [Fact]
+        public static void GetBytes_NegativeLength()
+        {
+            using (var deriveBytes = new PasswordDeriveBytes(TestPassword, s_testSalt))
+            {
+                Assert.Throws<OverflowException>(() => deriveBytes.GetBytes(-1));
+                Assert.Throws<OverflowException>(() => deriveBytes.GetBytes(int.MinValue));
+                Assert.Throws<OverflowException>(() => deriveBytes.GetBytes(int.MinValue / 2));
+            }
+        }
+
+        [Fact]
+        public static void GetBytes_NotIdempotent()
+        {
+            byte[] first;
+            byte[] second;
+
+            using (var deriveBytes = new PasswordDeriveBytes(TestPassword, s_testSalt))
+            {
+                first = deriveBytes.GetBytes(32);
+                second = deriveBytes.GetBytes(32);
+            }
+
+            Assert.NotEqual(first, second);
+        }
+
+        [Fact]
+        public static void GetBytes_IdempotentIfReset()
```

Comment:
I'd go with "StableIfReset", since Reset doesn't make the function become Idempotent